protostar/starknet_gateway/gateway_facade.py

```diff
@@ -212,7 +210,7 @@ async def _create_declare_tx_v1(
         )
         # pylint: disable=protected-access
         max_fee = await account_client._get_max_fee(
-            transaction=declare_tx, max_fee=max_fee, auto_estimate=auto_estimate_fee
+            transaction=declare_tx, max_fee=max_fee, auto_estimate=max_fee is None
```

Comment:
```suggestion
            transaction=declare_tx, max_fee=max_fee, auto_estimate=max_fee == "auto"
```
max_fee should always be provided to protect from unexpected costs. `max_fee` will be required arguments once we drop support for transactions v0

Reply:
[change type of max_fee to Fee](https://github.com/software-mansion/protostar/pull/979/commits/8b3d0ca5d7f7c558635efaccc2bc6c78e0f11c82)

Reply:
This is an external call and we have to operate on the type that's there: `max_fee: Optional[int] = None,` or did I miss something?

Reply:
I'm not sure if I get your point. The `Fee` type can be adapted to Starknet.py interface.

Reply:
I understand, but in this case, we'd first need it there in the starknet.py as it is an external lib to the protostar, right? And only then update the version in the protostar and update the code here accordingly. Am I correct?

Reply:
I mean, we could make `max_fee` required in `GatewayFacade::_create_declare_tx_v1`. Anyway, it's a private method. Resolving.

Reply:
It is required now in `_create_declare_tx_v1`